highlighters/main/main-highlighter.zsh

```diff
@@ -104,7 +104,8 @@ _zsh_highlight_main_add_region_highlight() {
   (( start -= $#PREBUFFER ))
   (( end -= $#PREBUFFER ))
 
-  (( end < 0 )) && return # having end<0 would be a bug
+  (( start >= end )) && { print -r -- >&2 'zsh-syntax-highlighting: BUG: _zsh_highlight_main_add_region_highlight: start >= end'; return }
```

Comment:
Suggest to interpolate `"$start"` and `"$end"` into the error message.

(Generally, most error message should have something interpolated into them.)